plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreAuthPgPassConfigurator.java

```diff
@@ -34,29 +38,62 @@
 public class PostgreAuthPgPassConfigurator implements IObjectPropertyConfigurator<Object, DBPDataSourceContainer> {
 
     protected Text usernameText;
+    protected Text overriddenHostnameText;
+    protected Button overrideHostname;
+    private Label hostnameLabel;
+
 
     @Override
     public void createControl(@NotNull Composite authPanel, Object object, @NotNull Runnable propertyChangeListener) {
         int fontHeight = UIUtils.getFontHeight(authPanel);
 
         Label usernameLabel = UIUtils.createLabel(authPanel, UIConnectionMessages.dialog_connection_auth_label_username);
-        usernameLabel.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING));
+        usernameLabel.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
 
         usernameText = new Text(authPanel, SWT.BORDER);
-        GridData gd = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+        overrideHostname = UIUtils.createCheckbox(authPanel, false);
+        overrideHostname.setText(UIConnectionMessages.dialog_connection_pgpass_hostname_override);
+        overrideHostname.setToolTipText(UIConnectionMessages.dialog_connection_pgpass_hostname_override_tip);
+        UIUtils.createEmptyLabel(authPanel, 1, 1);
+        hostnameLabel = UIUtils.createLabel(authPanel, "Hostname");
+        hostnameLabel.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+        overriddenHostnameText = new Text(authPanel, SWT.BORDER);
+        overrideHostname.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                overriddenHostnameText.setEnabled(overrideHostname.getSelection());
+                hostnameLabel.setEnabled(overrideHostname.getSelection());
+            }
+        });
+
+        GridData gd = new GridData(GridData.FILL_HORIZONTAL);
         gd.widthHint = fontHeight * 20;
+        overriddenHostnameText.setLayoutData(gd);
         usernameText.setLayoutData(gd);
         usernameText.addModifyListener(e -> propertyChangeListener.run());
+        authPanel.layout();
     }
 
     @Override
     public void loadSettings(@NotNull DBPDataSourceContainer dataSource) {
+        this.overrideHostname.setSelection(!CommonUtils.isEmpty(dataSource.getConnectionConfiguration().getProviderProperty(
+            PostgreConstants.PG_PASS_HOSTNAME)));
+        overriddenHostnameText.setEnabled(overrideHostname.getSelection());
+        hostnameLabel.setEnabled(overrideHostname.getSelection());
+        if (overrideHostname.getSelection()) {
+            this.overrideHostname.setText(dataSource.getConnectionConfiguration().getProviderProperty(PostgreConstants.PG_PASS_HOSTNAME));
+        }
         this.usernameText.setText(CommonUtils.notEmpty(dataSource.getConnectionConfiguration().getUserName()));
```

Comment:
Please extract connection configuration to a local variable and use `if` to avoid having this mess.